cpp/src/arrow/compute/api_scalar.h

```diff
@@ -90,6 +90,36 @@ class ARROW_EXPORT RoundOptions : public FunctionOptions {
   RoundMode round_mode;
 };
 
+enum class CalendarUnit : int8_t {
+  NANOSECOND,
+  MICROSECOND,
+  MILLISECOND,
+  SECOND,
+  MINUTE,
+  HOUR,
+  DAY,
+  WEEK,
+  MONTH,
+  SEASON,
```

Comment:
Can we define exactly what SEASON is?

Reply:
I'm not exactly sure myself. Here it [looks like quarter](https://github.com/tidyverse/lubridate/blob/9cece5680d0280c5cd1ccdbc9309ccb4724b3a34/R/round.r#L199-L201) but from looking at output it seems to be shifted a month back (Winter runs Dec 1 - March 1 etc) and it's not stable - first season start moves depending on the multiple.
I'll just add quarter and we can put this in if desired later.